packages/flutter/lib/src/material/search.dart

```diff
@@ -68,11 +68,13 @@ Future<T?> showSearch<T>({
   required SearchDelegate<T> delegate,
   String? query = '',
   bool useRootNavigator = false,
+  bool maintainState = false,
 }) {
   delegate.query = query ?? delegate.query;
   delegate._currentBody = _SearchBody.suggestions;
   return Navigator.of(context, rootNavigator: useRootNavigator).push(_SearchPageRoute<T>(
     delegate: delegate,
+    maintainStateOverride: maintainState,
```

Comment:
Instead of calling the new property `maintainStateOverride`, why not have the property annotated with an override annotation?

dart```
class _SearchPageRoute<T> extends PageRoute<T> {

// ...

@override
final bool maintainState;

// ...
```

Reply:
Thank you for that tip.

I'm not very familiar with dart tests 😞, figuring out how to write a test for this.